addons/mail/models/ir_model.py

```diff
@@ -100,3 +100,31 @@ def _instanciate(self, model_data):
             parents = [parents] if isinstance(parents, str) else parents
             model_class._inherit = parents + ['mail.thread.blacklist']
         return model_class
+
+    def _get_model_definitions(self, model_names_to_fetch):
+        response = {}
+        for model_name in model_names_to_fetch:
+            # get model fields, relational fields are kept only if the
+            # corresponding model is in model_names_to_fetch
+            model_fields = {
+                fname: field
+                for fname, field in self.env[model_name].fields_get(
+                    attributes=['name', 'type', 'relation', 'required', 'readonly', 'selection', 'string']).items()
+                if field.get('name') and (not field.get('relation') or field['relation'] in model_names_to_fetch)
+            }
+            # get default values for the model fields, exclude those that
+            # are not json serializable
+            fnames_to_default_values = self.env[model_name].default_get([
+                field['name'] for field in model_fields.values()
+                if field['type'] not in ['binary', 'date', 'datetime']
+            ])
```

Comment:
The problem is that some defaults are dynamic, even if they are serializable, if they are the result of a function it might not work as expected.

For example a date is easily serializable, but if it's the result of "now", we can't use that value.

So for defaults it might be more appropriate to dynamically call the server in real time during the tests to get them, rather than doing it here.

Reply:
Probably for now, only return defaults that are hard-code values and not a function name or the result of a function?